frontend/simpleRedux.js

```diff
@@ -5,6 +5,102 @@ const {CARDANOLITE_CONFIG} = require('./frontendConfigLoader')
 let state, middlewares
 const root = {}
 
+const createReverseArray = (nodeChildren) => {
+  const ret = []
+  for (let i = nodeChildren.length - 1; i >= 0; i--) {
+    ret.push(nodeChildren[i])
+  }
+  return ret
+}
+
+const reconcile = (existingNode, virtualNode) => {
+  if (existingNode.tagName !== virtualNode.tagName) return virtualNode
+
+  // arrays reversed so that we can process children from their tail using pop
+  const [existingChildren, virtualChildren] = [existingNode.children, virtualNode.children].map(
+    (children) => createReverseArray(children).filter((n) => n instanceof HTMLElement)
+  )
+
+  // attributes is a NamedNodeMap, which does not support same api as an array
+  for (let i = 0; i < virtualNode.attributes.length; i++) {
+    if (
+      existingNode.getAttribute(virtualNode.attributes[i].name) !== virtualNode.attributes[i].value
+    ) {
+      existingNode.setAttribute(virtualNode.attributes[i].name, virtualNode.attributes[i].value)
```

Comment:
Setting attributes like this works - I don't know about performance, but I don't see other way around it, apart for skipping those that did not change (now done in the last commit).